addons/hr_holidays/models/hr_leave.py

```diff
@@ -668,9 +668,17 @@ def _check_validity(self):
                     continue
                 if previous_emp_data != emp_data and len(emp_data) >= len(previous_emp_data):
                     raise ValidationError(_("There is no valid allocation to cover that request."))
-        is_leave_user = self.env.user.has_group('hr_holidays.group_hr_holidays_user')
-        if not is_leave_user and any(leave.has_mandatory_day for leave in self):
-            raise ValidationError(_('You are not allowed to request time off on a Mandatory Day'))
+        is_leave_user = self.env.user.has_group('hr_holidays.group_hr_holidays_user') or all(
+            leave.employee_id.leave_manager_id
+            and leave.employee_id.leave_manager_id.id == self.env.uid
+            for leave in self
+        )
+        if (
+            not is_leave_user
+            and any(leave.has_mandatory_day for leave in self)
+            and not self.env.context.get("skip_mandatory_days_check", False)
+        ):
```

Comment:
Also I think this more readable 
```suggestion
        if (not is_leave_user
            and any(leave.has_mandatory_day for leave in self)
            and not self.env.context.get("skip_mandatory_days_check", False)):
```